portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +24,35 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    // instance vairables 
+    ArrayList<String> messages = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("Hello Fran you did it you solved!!");
+
+    //make sure that if the page is reloaded the messages our reset
+    messages.clear();
```

Comment:
Why store messages on the class if it's:
 - reset every function call
 - Not used in any other function?

Reply:
I was clearing it because sometimes I accidentally loaded the page twice and my messages would be added in as more times than necessary. Additionally, I looked at the technical chat because I was pretty confused where to initialize/declare the Arraylist and many other participants declared it in the exact same manner (I guess they didnt have the duplicate error). But I will look it into it for the next PR and see if i make a separate function I call to handle this

Reply:
I would look at this like a lifetime decision. There's 2 important things to keep in mind:
 1) The server lifetime
 2) The lifetime of the request.

Currently, you have the messages var tied to the server lifetime. (Since it's a member of the class,) but, it's modified for every request. (Hence why double messages were showing up). In terms of solutions, I would either:
 - Change the lifetime of "messages" to be that of request. (Eg Declare and initialize the var inside the doFn)
 - Or (better option!) Change the location of the modification to match the server lifetime. So instead of populating messages in the doFn, do so in the server init() fn. (Which you'd have to add). init() will be called when the server starts, so by the time the doFn executes, you'll have a fully populated messages!